meson.build

```diff
@@ -0,0 +1,484 @@
+)
+
+
+# libxkbcommon.
+# Note: we use some yacc extensions, which work with either GNU bison
+# (preferred) or byacc. Other yacc's may or may not work.
+yacc = find_program('bison', 'byacc')
+yacc_gen = generator(
+    yacc,
+    output: ['@BASENAME@.c', '@BASENAME@.h'],
+    arguments: ['@INPUT@', '--defines=@OUTPUT1@', '--output=@OUTPUT0@', '-p _xkbcommon_'],
+)
+libxkbcommon_internal = static_library(
+    'xkbcommon-internal',
+    'src/compose/parser.c',
+    'src/compose/parser.h',
+    'src/compose/paths.c',
+    'src/compose/paths.h',
+    'src/compose/state.c',
+    'src/compose/table.c',
+    'src/compose/table.h',
+    'src/xkbcomp/action.c',
+    'src/xkbcomp/action.h',
+    'src/xkbcomp/ast.h',
+    'src/xkbcomp/ast-build.c',
+    'src/xkbcomp/ast-build.h',
+    'src/xkbcomp/compat.c',
+    'src/xkbcomp/expr.c',
+    'src/xkbcomp/expr.h',
+    'src/xkbcomp/include.c',
+    'src/xkbcomp/include.h',
+    'src/xkbcomp/keycodes.c',
+    'src/xkbcomp/keymap.c',
+    'src/xkbcomp/keymap-dump.c',
+    'src/xkbcomp/keywords.c',
+    yacc_gen.process('src/xkbcomp/parser.y'),
+    'src/xkbcomp/parser-priv.h',
+    'src/xkbcomp/rules.c',
+    'src/xkbcomp/rules.h',
+    'src/xkbcomp/scanner.c',
+    'src/xkbcomp/symbols.c',
+    'src/xkbcomp/types.c',
+    'src/xkbcomp/vmod.c',
+    'src/xkbcomp/vmod.h',
+    'src/xkbcomp/xkbcomp.c',
+    'src/xkbcomp/xkbcomp-priv.h',
+    'src/atom.c',
+    'src/atom.h',
+    'src/context.c',
+    'src/context.h',
+    'src/context-priv.c',
+    'src/darray.h',
+    'src/keysym.c',
+    'src/keysym.h',
+    'src/keysym-utf.c',
+    'src/ks_tables.h',
+    'src/keymap.c',
+    'src/keymap.h',
+    'src/keymap-priv.c',
+    'src/scanner-utils.h',
+    'src/state.c',
+    'src/text.c',
+    'src/text.h',
+    'src/utf8.c',
+    'src/utf8.h',
+    'src/utils.c',
+    'src/utils.h',
+    include_directories: include_directories('src'),
+)
+libxkbcommon_link_args = []
+if have_version_script
+    libxkbcommon_link_args += '-Wl,--version-script=' + join_paths(meson.source_root(), 'xkbcommon.map')
+endif
+libxkbcommon = library(
+    'xkbcommon',
+    'xkbcommon/xkbcommon.h',
+    link_whole: libxkbcommon_internal,
+    link_args: libxkbcommon_link_args,
+    link_depends: 'xkbcommon.map',
+    version: '0.0.0',
+    install: true,
+)
+install_headers(
+    'xkbcommon/xkbcommon.h',
+    'xkbcommon/xkbcommon-compat.h',
+    'xkbcommon/xkbcommon-compose.h',
+    'xkbcommon/xkbcommon-keysyms.h',
+    'xkbcommon/xkbcommon-names.h',
+    subdir: 'xkbcommon',
+)
+pkgconfig.generate(
+    name: 'xkbcommon',
+    filebase: 'xkbcommon',
+    libraries: libxkbcommon,
+    version: meson.project_version(),
+    description: 'XKB API common to servers and clients',
+)
+
+
+# libxkbcommon-x11.
+if get_option('enable-x11')
+    xcb_dep = dependency('xcb', version: '>=1.10', required: false)
+    xcb_xkb_dep = dependency('xcb-xkb', version: '>=1.10', required: false)
+    if not xcb_dep.found() or not xcb_xkb_dep.found()
```

Comment:
Yes, it's for the error message.

The component idea definitely looks like it would fit here.